docs/source/srs.rst

```diff
@@ -0,0 +1,283 @@
+proper initialization and configuration before it will collect data. The
+following procedure is informed by the BNO055's datasheet which details
+correct usage of the chip. After establishing I2C connection with the chip,
+the driver class will:
+
+* check that the BNO055's self-test initiated at startup was successful
+* switch the BNO055 from configuration mode to NDOF operation mode
+
+The NDOF operation mode calculates fused absolute orientation data from the
+accelerometer, gyroscope and magnetometer. More information can be found in
+the BNO055 datasheet.
+
+Notification of BNO055 Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The BNO055 will collect the following data in 3D space (X,Y,Z):
+
+* Euler angles
+* gyroscope measurements
+* linear acceleration
+* accelerometer measurements
+
+The data is broadcast on the CANopen network for other boards to read.
+
+User Classes and Characteristics
+--------------------------------
+
+The IMU chip is an auxiliary board that does not require any user interaction
+post-development?
+
+Operating Environments
+----------------------
+The software will operate on the ST microcontroller present on the DEV1 IMU.
+The software environment is embedded with no operating system present. All
+development will take place through the EVT-core library and will interact
+directly with the ST microcontroller.
+
+User Documentation
+------------------
+
+Additional documentation will need to exist for boards that will react to
+the DEV1 IMU's reported data. Information for each board will need to include
+which data it will use and how it will adjust itself accordingly. This will
+provide users with a holistic view of how the DEV1 IMU is used in the bike.
+
+Constraints
+-----------
+
+Below are some constraints worth considering. They are a fixed part of the
+system.
+
+* Development must be in C/C++
+* Communication will take place using CANopen
+* EVT-core will be used for low level microcontroller interfacing
+* Must be developed for the STM32F334r8
+* Resulting binary must fit within the ST microcontroller 64KB flash memory
+* Orientation measurements must be performed by the BNO055
+* non standard I2C communication with the BNO055
+
+Assumptions and Dependencies
+----------------------------
+
+It is assumed that the BNO055 chip will behave exactly as described by its
+datasheet, unless communication between the BNO055 chip and ST
+microcontroller fails.
+
+Apportioning of Requirements
+----------------------------
+
+At this point in the life cycle of the DEV1 IMU project, deployment has begun.
+Due to the nature of student-run teams, it is unlikely that there will be
+further revision to this system, as new students will likely start new projects.
+As a result, there are no future requirements planned to be added to this
+system.
+
+Specific Requirements
+=====================
+
+External Interface Requirements
+-------------------------------
+
+* no external interface?
```

Comment:
fix!

Reply:
Question for @mjmagee991 - If I'm right, since the IMU chip is only broadcasting data, there is no need for any external interface?